actors/market/tests/market_actor_test.rs

```diff
@@ -452,22 +391,24 @@ fn expect_provider_control_address(
 
 fn add_provider_funds(
     rt: &mut MockRuntime,
+    amount: TokenAmount,
     provider: Address,
     owner: Address,
     worker: Address,
-    amount: TokenAmount,
 ) {
     rt.set_value(amount.clone());
-
+    // TODO: call rt.SetAddressActorType(minerAddrs.provider, builtin.StorageMinerActorCodeID)?
```

Comment:
I would check provider's code ID, if set assert miner, if not set it to miner.  Whether you do this or NOP the TODO should go away before merging.

Reply:
I would do this in a sub function that takes a map for merging into the rt addr:code-id map. You can then share this between add_provider_funds and the other add balance and withdraw helpers that also hit the same code path.

Reply:
Not sure if I understand exactly what you wanted. I'm more 1:1 faithfull to the Go code now. Please let me know.

Reply:
This is all good

Reply:
Note on review process: if there are only minor fixes as in this case I'll leave your PR approved so that you can merge it yourself once you've addressed all feedback.  You can reach out here or on filecoin slack to ask for clarification like this to be sure we're talking about the same thing before merging.  You can also re-request me for GitHub review and I'll do another pass.